app/components/person-search-by-name.js

```diff
@@ -26,20 +26,28 @@ export default class PersonSearchByNameComponent extends Component {
       dataMapping: (data) => {
         const entry = data.attributes;
         return {
-          given_name: entry.given_name,
-          family_name: entry.family_name,
+          given_name: entry.given_name.trim().toLowerCase(),
+          family_name: entry.family_name.trim().toLowerCase(),
         };
       },
     });
-
-    result = new Set(result.toArray());
+    result = result
+      .toArray()
+      .filter(
+        (v, i, a) =>
+          a.findIndex((v2) =>
+            ['family_name', 'given_name'].every((k) => v2[k] === v[k])
+          ) === i
+      );
```

Comment:
Can you maybe add a comment about what this filter does ? It's not straightforward when reading it :)

Reply:
indeed! I extracted the code in a separate function and added a documentation to it